Atomic/sign.py

```diff
@@ -0,0 +1,89 @@
+from . import util
+from . import Atomic
+import os
+import tempfile
+from .atomic import AtomicError
+from .client import ids_same, add_shaw
+
+
+class Sign(Atomic):
+    def sign(self):
+        # TODO
+        # If an alternate sign key is provided, should we verify that it is valid?
+        # Should we check to see if there are valid GPG signatures?
+        # Atomic is run as sudo. Should we work around that?
+
+        if self.args.debug:
+            util.write_out(str(self.args))
+
+        # Verify all the images provided are valid
+        self.check_input_validity()
+
+        signature_path = util.get_signature_path(self.atomic_config)
+
+        signer = util.get_atomic_config_item(['default_signer']) if \
+            self.args.sign_key is None else self.args.sign_key
+
+        # Make sure the signature path exists
+        if not os.path.exists(signature_path):
+            raise ValueError("The path signature path {} "
+                             "does not exist".format(signature_path))
+
+        for sign_image in self.args.sign_targets:
+            local_inspect_info = self._inspect_image(image=sign_image)
```

Comment:
One of the working assumptions for creating a local signature was that the image would also have to be local.